### PR TITLE
[TS migration] Migrate 'ReportActionItemMoneyRequest' component to TypeScript

### PR DESCRIPTION
**Pull request by VickyStash**
### Details

[TS migration] Migrate 'ReportActionItemMoneyRequest' component to TypeScript

### Fixed Issues

$ https://github.com/Expensify/App/issues/31983
PROPOSAL: N/A

### Tests

- [x] Verify that no errors appear in the JS console

1. Open app
2. Log in with any account that has a workspace

**Requests to another User**
1. Create several `Manual` and `Scan` money requests to another user.
2. Open the chat with that user and click on the grouped requests report preview.
3. Make sure the money requests preview looks as expected. They have the correct amount and currency. If the Scan request is still in progress, the message should say `Receipt scan in progress...`.
4. Tap on `Manual` Money request preview to open its view. Make sure it looks as expected and has all the necessary data. You can also update Description/Date fields.
5. Tap on `Scan` Money request preview to open its view. Make sure it looks same way as before and has all the necessary data.

**Requests to Workspace**
1. Create several `Manual`,`Scan` and `Distance` money requests to the workspace.
2. Open the workspace chat and click on the grouped requests report preview.
3. Make sure the money requests preview looks as expected. The Distance request preview should have a map (or default image if the request is still processed) with a distance, currency and amount of money as well as the distance value.
4. Tap on `Distance` Money request preview to open its view. Make sure it looks the same way as before and has all the necessary data.

### Offline tests

1. Open app
2. Log in with any account that has a workspace
3. Create several `Manual`,`Scan` and `Distance` money requests to the workspace.
4. Open the workspace chat and click on the grouped requests report preview.
5. Make sure the money requests preview looks as expected. 
6. Go offline
7. Delete a money request
8. Make sure that the request skeleton shows the correct amount and currency.

### QA Steps

- [x] Verify that no errors appear in the JS console

Same as in **Tests** + **Offline Tests** sections

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/23176449/be7082e3-86be-41ee-af90-c8276d393f9e

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/23176449/0402b40b-f4a4-4d38-a552-52af63b44e1a

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/23176449/76f162af-bd9d-4310-93af-74fc02e76fb7

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/23176449/b1fb7daa-5a1f-4a7b-b451-78b5c5cc7936

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/23176449/93d67aef-ce6b-431c-abb2-5a438e4a3dfe

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/23176449/b13c826c-3b33-49e2-9774-a5c6f3621359

</details>

### CODE REVIEW

File: src/components/ReportActionItem/MoneyRequestPreview.tsx [hunk 1]
```diff
@@ -229,18 +219,18 @@ function MoneyRequestPreview(props) {
...
             return Localize.translateLocal('iou.receiptMissingDetails');
         }
 
         return CurrencyUtils.convertToDisplayString(requestAmount, requestCurrency);
     };
 
-    const getDisplayDeleteAmountText = () => {
-        const {amount, currency} = ReportUtils.getTransactionDetails(props.action.originalMessage);
+    const getDisplayDeleteAmountText = (): string => {
+        const {amount, currency} = ReportUtils.getTransactionDetails(action.originalMessage as OnyxTypes.Transaction) ?? {};
```
Comment (VickyStash) [inline src/components/ReportActionItem/MoneyRequestPreview.tsx:230]: I have some doubts regarding this assertion
1. `action.originalMessage` looks like this
image
2. Updating the type of `ReportUtils.getTransactionDetails` to accept `IOUMessage` type as well will trigger a lot of updates

Comment (fabioh8010) [reply src/components/ReportActionItem/MoneyRequestPreview.tsx:232]: This is very very strange indeed, `IOUMessage` and `Transaction` looks like completely different things. Maybe we could create another function similar to what `getTransactionDetails` do, but to only return `amount` and `currency`?

File: src/components/ReportActionItem/MoneyRequestPreview.tsx [hunk 2]
```diff
@@ -232,18 +221,18 @@ function MoneyRequestPreview(props) {
...
 
         if (isDistanceRequest) {
-            return CurrencyUtils.convertToDisplayString(TransactionUtils.getAmount(props.action.originalMessage), currency);
+            return CurrencyUtils.convertToDisplayString(TransactionUtils.getAmount(action.originalMessage as OnyxTypes.Transaction), currency);
```
Comment (fabioh8010) [inline src/components/ReportActionItem/MoneyRequestPreview.tsx:235]: Same here I guess

File: src/components/ReportActionItem/MoneyRequestView.tsx [hunk 3]
```diff
@@ -288,80 +259,82 @@ function MoneyRequestView({report, parentReport, parentReportActions, policyCate
...
                         wrapperStyle={[styles.pv2, styles.taskDescriptionMenuItem]}
-                        brickRoadIndicator={hasViolations('comment') ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : ''}
+                        brickRoadIndicator={hasViolations('comment') ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : undefined}
                         numberOfLinesTitle={0}
                     />
                     {canUseViolations && <ViolationMessages violations={getViolationsForField('comment')} />}
                 </OfflineWithFeedback>
                 {isDistanceRequest ? (
-                    <OfflineWithFeedback pendingAction={lodashGet(transaction, 'pendingFields.waypoints') || lodashGet(transaction, 'pendingAction')}>
+                    <OfflineWithFeedback pendingAction={transaction?.pendingFields?.waypoints ?? transaction?.pendingAction}>
```
Comment (fabioh8010) [inline src/components/ReportActionItem/MoneyRequestView.tsx:268]: Can we use `getPendingFieldAction` here too?

File: src/components/ReportActionItem/MoneyRequestView.tsx [hunk 4]
```diff
@@ -288,80 +259,82 @@ function MoneyRequestView({report, parentReport, parentReportActions, policyCate
...
                         onPress={() => Navigation.navigate(ROUTES.EDIT_REQUEST.getRoute(report.reportID, CONST.EDIT_REQUEST_FIELD.DATE))}
-                        brickRoadIndicator={hasViolations('date') || (hasErrors && transactionDate === '') ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : ''}
+                        brickRoadIndicator={hasViolations('date') || (hasErrors && transactionDate === '') ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : undefined}
                         error={hasErrors && transactionDate === '' ? translate('common.error.enterDate') : ''}
                     />
                     {canUseViolations && <ViolationMessages violations={getViolationsForField('date')} />}
                 </OfflineWithFeedback>
                 {shouldShowCategory && (
-                    <OfflineWithFeedback pendingAction={lodashGet(transaction, 'pendingFields.category') || lodashGet(transaction, 'pendingAction')}>
+                    <OfflineWithFeedback pendingAction={transaction?.pendingFields?.category ?? transaction?.pendingAction}>
```
Comment (fabioh8010) [inline src/components/ReportActionItem/MoneyRequestView.tsx:309]: Same, can we use `getPendingFieldAction`?

File: src/components/ReportActionItem/MoneyRequestView.tsx [hunk 5]
```diff
@@ -288,80 +259,82 @@ function MoneyRequestView({report, parentReport, parentReportActions, policyCate
...
                             onPress={() => Navigation.navigate(ROUTES.EDIT_REQUEST.getRoute(report.reportID, CONST.EDIT_REQUEST_FIELD.CATEGORY))}
-                            brickRoadIndicator={hasViolations('category') ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : ''}
+                            brickRoadIndicator={hasViolations('category') ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : undefined}
                         />
                         {canUseViolations && <ViolationMessages violations={getViolationsForField('category')} />}
                     </OfflineWithFeedback>
                 )}
                 {shouldShowTag && (
-                    <OfflineWithFeedback pendingAction={lodashGet(transaction, 'pendingFields.tag') || lodashGet(transaction, 'pendingAction')}>
+                    <OfflineWithFeedback pendingAction={transaction?.pendingFields?.tag ?? transaction?.pendingAction}>
```
Comment (fabioh8010) [inline src/components/ReportActionItem/MoneyRequestView.tsx:323]: Same

File: src/components/ReportActionItem/MoneyRequestView.tsx [hunk 6]
```diff
@@ -397,50 +370,42 @@ function MoneyRequestView({report, parentReport, parentReportActions, policyCate
...
+            const parentReportAction = ReportActionsUtils.getParentReportAction(report);
+            const originalMessage = parentReportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU ? parentReportAction.originalMessage : undefined;
+            const transactionID = originalMessage?.IOUTransactionID ?? 0;
+            return `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`;
         },
-    }),
-    withOnyx({
+    },
+})(
+    withOnyx<MoneyRequestViewProps, MoneyRequestViewTransactionOnyxProps>({
```
Comment (fabioh8010) [inline src/components/ReportActionItem/MoneyRequestView.tsx:401]: Why did you move `transactionViolations` and `policyTags` to the other HOC? Isn't possible to keep it here?

Comment (VickyStash) [reply src/components/ReportActionItem/MoneyRequestView.tsx:399]: It's possible, but I don't see a reason. The second HOC exists to use data from the first HOC, so transactions use `parentReportActions`, that's why `transactions` are in the second HOC, but `transactionViolations` and `policyTags` doesn't use any data from the first HOC

File: src/components/ReportActionItem/MoneyRequestPreview.tsx (hunk 1 above)
Comment (VickyStash) [reply src/components/ReportActionItem/MoneyRequestPreview.tsx:232]: I've tested it more, and I guess the function can be updated to
```
    const getDisplayDeleteAmountText = (): string => {
        const iouOriginalMessage: IOUMessage | EmptyObject = action?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU ? action.originalMessage : {};
        const {amount = 0, currency = CONST.CURRENCY.USD} = iouOriginalMessage;

        return CurrencyUtils.convertToDisplayString(amount, currency);
    };
```
What do you think?

File: src/components/ReportActionItem/MoneyRequestPreview.tsx [hunk 7]
```diff
@@ -232,19 +223,16 @@ function MoneyRequestPreview(props) {
...
-
-        if (isDistanceRequest) {
-            return CurrencyUtils.convertToDisplayString(TransactionUtils.getAmount(props.action.originalMessage), currency);
-        }
+    const getDisplayDeleteAmountText = (): string => {
+        const iouOriginalMessage: IOUMessage | EmptyObject = action?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU ? action.originalMessage : {};
+        const {amount = 0, currency = CONST.CURRENCY.USD} = iouOriginalMessage;
 
         return CurrencyUtils.convertToDisplayString(amount, currency);
     };
```
Comment (VickyStash) [inline src/components/ReportActionItem/MoneyRequestPreview.tsx:238]: During this file migration, it turned out that `originalMessage` of action is passed to methods where `transaction` is expected to get the amount and currency values. Though `originalMessage` doesn't look like transaction at all
image
After some testing I've find out that the functionality of `getDisplayDeleteAmountText` function can be simplified.

**Review by fabioh8010 — APPROVED** [review]

**Review by blazejkustra — APPROVED** [review]
Great work! LGTM

**Comment by trevor-coleman** [comment]
#33969 is depending on this, commenting so I get notified when it gets merged!

**Comment by VickyStash** [comment]
@rushatgabhane Hey, any ETA on reviewing this one? 🙂

**Comment by rushatgabhane** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by rushatgabhane — APPROVED** [review]
LGTM

**Comment by fabioh8010** [comment]
@mountiny Could we assign a engineer here? Looks like Melvin didn't assign anyone.
@VickyStash Please fix conflicts 🙂

**Comment by rushatgabhane** [comment]
@aldo-expensify all yours!

**Comment by aldo-expensify** [comment]
@VickyStash there are conflicts now, please resolve 🙏

**Comment by VickyStash** [comment]
@aldo-expensify Done 👌

**Review by aldo-expensify — APPROVED** [review]

**Comment by aldo-expensify** [comment]
Thank you!, this is a huge PR, I skim read the changes and didn't see anything wrong, I hope everything is fine 🥲

File: src/components/ReportActionItem/MoneyRequestPreview.tsx [hunk 8]
```diff
@@ -317,31 +301,31 @@ function MoneyRequestPreview(props) {
...
                                 <View style={[styles.flex1]}>
-                                    {!isCurrentUserManager && props.shouldShowPendingConversionMessage && (
+                                    {!isCurrentUserManager && shouldShowPendingConversionMessage && (
                                         <Text style={[styles.textLabel, styles.colorMuted]}>{translate('iou.pendingConversionMessage')}</Text>
                                     )}
                                     {shouldShowDescription && <RenderHTML html={parser.replace(merchantOrDescription)} />}
                                     {shouldShowMerchant && <Text style={[styles.textLabelSupporting, styles.textNormal]}>{merchantOrDescription}</Text>}
                                 </View>
-                                {props.isBillSplit && !_.isEmpty(participantAccountIDs) && requestAmount > 0 && (
+                                {isBillSplit && participantAccountIDs.length > 0 && requestAmount && requestAmount > 0 && (
```
Comment (situchan) [inline src/components/ReportActionItem/MoneyRequestPreview.tsx:324]: @VickyStash Native app crashes on this line. Can you please raise quick PR fixing that?
```suggestion
                                {isBillSplit && participantAccountIDs.length > 0 && !!requestAmount && requestAmount > 0 && (
```
Screenshot 2024-02-06 at 9 15 57 PM

Comment (VickyStash) [reply src/components/ReportActionItem/MoneyRequestPreview.tsx:324]: @situchan I'm taking a look

Comment (VickyStash) [reply src/components/ReportActionItem/MoneyRequestPreview.tsx:324]: @situchan could you please provide me steps for reproduction so I can be sure the issue is fixed?

Comment (situchan) [reply src/components/ReportActionItem/MoneyRequestPreview.tsx:324]: I tested that solution myself. Works well.
I was just testing https://github.com/Expensify/App/pull/35876. Please try to follow QA step on that PR.

Comment (situchan) [reply src/components/ReportActionItem/MoneyRequestPreview.tsx:324]: So that happens when `requestAmount` = 0
Try this:
A request $10 from B
B request $10 (same amount) from A

Comment (VickyStash) [reply src/components/ReportActionItem/MoneyRequestPreview.tsx:324]: @situchan I got what you mean and it makes sense to me, but I'm still not able to trigger this case on the main, could you please provide me reproduction video 🙏

File: src/components/ReportActionItem/MoneyRequestAction.tsx [hunk 9]
```diff
@@ -87,31 +70,32 @@ function MoneyRequestAction({
...
         if (isSplitBillAction) {
-            const reportActionID = lodashGet(action, 'reportActionID', '0');
+            const reportActionID = action.reportActionID ?? '0';
             Navigation.navigate(ROUTES.SPLIT_BILL_DETAILS.getRoute(chatReportID, reportActionID));
             return;
         }
 
         // If the childReportID is not present, we need to create a new thread
-        const childReportID = lodashGet(action, 'childReportID', 0);
+        const childReportID = action?.childReportID ?? '0';
```
Comment (youssef-lr) [inline src/components/ReportActionItem/MoneyRequestAction.tsx:95]: @VickyStash please remember that all report IDs must be numbers, this created this bug https://github.com/Expensify/App/issues/35904

Comment (youssef-lr) [inline src/components/ReportActionItem/MoneyRequestAction.tsx:95]: @VickyStash heads up! that all IDs must be numbers, this created this bug https://github.com/Expensify/App/issues/35904. :)

Comment (aldo-expensify) [reply src/components/ReportActionItem/MoneyRequestAction.tsx:95]: @youssef-lr reportIDs are expected to be string, otherwise they can overflow, right?

Comment (youssef-lr) [reply src/components/ReportActionItem/MoneyRequestAction.tsx:95]: Yeah sorry my bad, they are all numbers in the backend, but in JS `reportIDs` & `reportActionIDs` are strings. In this specific example it doesn't make sense to use `'0'`, because the condition below this line evaluates to `false` when it should evaluate to `true` `if (!childReportID) {`

Comment (VickyStash) [reply src/components/ReportActionItem/MoneyRequestAction.tsx:95]: @youssef-lr you are right, my bad!

Comment (aldo-expensify) [reply src/components/ReportActionItem/MoneyRequestAction.tsx:95]: Commented in your PR @youssef-lr , I don't think we should default to 0 either

Comment (youssef-lr) [reply src/components/ReportActionItem/MoneyRequestAction.tsx:95]: Yeah that sounds good, updated.
